src/Microsoft.Fast.Components.FluentUI/DesignTokens/DesignToken.razor.cs

```diff
@@ -0,0 +1,143 @@
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Components;
+using Microsoft.Extensions.Configuration;
+using Microsoft.JSInterop;
+
+namespace Microsoft.Fast.Components.FluentUI.DesignTokens;
+
+public partial class DesignToken<T> : ComponentBase, IDesignToken<T>, IAsyncDisposable
+{
+    [Inject]
+    private IJSRuntime JSRuntime { get; set; } = default!;
+
+    [Inject]
+    private IConfiguration Configuration { get; set; } = default!;
+
+    private Lazy<Task<IJSObjectReference>>? moduleTask;
+    private Reference Target { get; set; } = new Reference();
+
+    private T? _defaultValue;
+
+    /// <summary>
+    /// Gets the name of this design token 
+    /// </summary>
+    public string? Name { get; init; }
+
+    /// <summary>
+    /// Gets or sets the value of the design token
+    /// </summary>
+    [Parameter]
+    public T? Value { get; set; }
+
+    /// <summary>
+    /// Gets or sets the content to apply this design token on
+    /// </summary>
+    [Parameter]
+    public RenderFragment<Reference>? ChildContent { get; set; }
+
+    /// <summary>
+    /// Constructs an instance of a DesignToken.
+    /// </summary>
+    public DesignToken()
+    {
+
+    }
+
+    /// <summary>
+    /// Constructs an instance of a DesignToken.
+    /// </summary>
+    public DesignToken(IJSRuntime jsRuntime, IConfiguration configuration)
+    {
+        JSRuntime = jsRuntime;
+        Configuration = configuration;
+
+        Initialize();
+    }
+
+    /// <inheritdoc/>
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender && Value is not null)
+        {
+            Initialize();
+
+            await SetValueFor(Target.Current, Value);
+            StateHasChanged();
+        }
+    }
+
+    private void Initialize()
+    {
+        string scriptSource = Configuration["FluentWebComponentsScriptSource"] ?? "https://cdn.jsdelivr.net/npm/@fluentui/web-components/dist/web-components.min.js";
+
+        moduleTask = new(() => JSRuntime.InvokeAsync<IJSObjectReference>(
+            "import", scriptSource).AsTask());
+    }
+
+    /// <summary>
+    /// Sets the default value of this token
+    /// </summary>
+    public DesignToken<T> WithDefault(T value)
+    {
+        _defaultValue = value;
+        return this;
+    }
+
+    /// <summary>
+    /// Sets the value of the for the associated <see cref="ElementReference"/> to the default value
+    /// </summary>
+    /// <param name="element">the associated <see cref="ElementReference"/> </param>
+    /// <exception cref="Exception"></exception>
+    public async ValueTask SetValueFor(ElementReference element)
```

Comment:
I'm not sure if this is specifically needed for Blazor support, but the foundation system always inherits the default and does not provide and overload like this. That is, we probably don't need to apply the default value to a specific element. I believe we should remove this function and only have the two parameter variant below.